visual-tests/remote-download/remote-download-test.cpp

```diff
@@ -0,0 +1,142 @@
+    mImageView1.SetBackgroundColor(Vector4(0.0f, 0.0f, 0.0f, 0.0f));
+    defaultWindow.Add(mImageView1);
+
+    mImageView2 = Toolkit::ImageView::New("http://static.midomi.com/s/s/images/000/000/000/000/212/651/88/520_000000000000212651881500x1500_72dpi_RGB_q70.jpg");
+    mImageView2.SetProperty( Dali::Actor::Property::NAME,"mImageView2");
+    mImageView2.SetProperty( Actor::Property::PARENT_ORIGIN,ParentOrigin::TOP_LEFT);
+    mImageView2.SetProperty( Actor::Property::ANCHOR_POINT,AnchorPoint::TOP_CENTER);
+    mImageView2.SetProperty(Actor::Property::POSITION_USES_ANCHOR_POINT, false);
+    mImageView2.SetProperty( Actor::Property::POSITION, Vector2(400, 100));
+    mImageView2.SetBackgroundColor(Vector4(0.0f, 0.0f, 0.0f, 0.0f));
+    defaultWindow.Add(mImageView2);
+
+    mImageView3 = Toolkit::ImageView::New("http://static.midomi.com/s/s/images/000/000/000/000/212/353/21/520_000000000000212353211500x1500_72dpi_RGB_q70.jpg");
+    mImageView3.SetProperty( Dali::Actor::Property::NAME,"mImageView3");
+    mImageView3.SetProperty( Actor::Property::PARENT_ORIGIN,ParentOrigin::TOP_LEFT);
+    mImageView3.SetProperty( Actor::Property::ANCHOR_POINT,AnchorPoint::TOP_CENTER);
+    mImageView3.SetProperty(Actor::Property::POSITION_USES_ANCHOR_POINT, false);
+    mImageView3.SetProperty( Actor::Property::POSITION, Vector2(0, 400));
+    mImageView3.SetBackgroundColor(Vector4(0.0f, 0.0f, 0.0f, 0.0f));
+    defaultWindow.Add(mImageView3);
+
+    mImageView4 = Toolkit::ImageView::New("http://d2k43l0oslhof9.cloudfront.net/platform/image/contents/vc/20/01/58/20170629100630071189_0bf6b911-a847-cba4-e518-be40fe2f579420170629192203240.jpg");
+    mImageView4.SetProperty( Dali::Actor::Property::NAME,"mImageView4");
+    mImageView4.SetProperty( Actor::Property::PARENT_ORIGIN,ParentOrigin::TOP_LEFT);
+    mImageView4.SetProperty( Actor::Property::ANCHOR_POINT,AnchorPoint::TOP_CENTER);
+    mImageView4.SetProperty(Actor::Property::POSITION_USES_ANCHOR_POINT, false);
+    mImageView4.SetProperty( Actor::Property::POSITION, Vector2(400, 400));
+    mImageView4.SetBackgroundColor(Vector4(0.0f, 0.0f, 0.0f, 0.0f));
+    defaultWindow.Add(mImageView4);
+
+    mImageView5 = Toolkit::ImageView::New("http://static.midomi.com/h/images/w/weather_sunny.png");
+    mImageView5.SetProperty( Dali::Actor::Property::NAME,"mImageView5");
+    mImageView4.SetProperty( Actor::Property::PARENT_ORIGIN,ParentOrigin::TOP_LEFT);
+    mImageView5.SetProperty( Actor::Property::ANCHOR_POINT,AnchorPoint::TOP_CENTER);
+    mImageView5.SetProperty(Actor::Property::POSITION_USES_ANCHOR_POINT, false);
+    mImageView5.SetProperty( Actor::Property::POSITION, Vector2(800, 100));
+    mImageView5.SetBackgroundColor(Vector4(0.0f, 0.0f, 0.0f, 0.0f));
+    defaultWindow.Add(mImageView5);
+
+    // Start the test
+    mTimer = Timer::New( 3000 ); // ms
+    mTimer.TickSignal().Connect( this, &RemoteDownloadTest::OnTimer);
+    mTimer.Start();
```

Comment:
Can you not wait for all 5 image views to have loaded to perform the test as this could potentially take longer than 3 seconds in some scenarios or may be much quicker in which case you'd be waiting pointlessly.